lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -119,6 +119,7 @@ def create
     t = Tempfile.new('openldap_database')
     t << "dn: olcDatabase=#{resource[:backend]},cn=config\n"
     t << "changetype: add\n"
+    t << "objectClass: olcDatabaseConfig\n"
```

Comment:
I still think my fix is valid, as the [documentation](http://www.openldap.org/doc/admin24/slapdconf2.html#Database-specific%20Directives) states that "[d]atabase entries must have the olcDatabaseConfig objectClass".